internals/secrethub/service_gcp_link.go

```diff
@@ -0,0 +1,193 @@
+package secrethub
+
+import (
+	"fmt"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/progress"
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+	"github.com/secrethub/secrethub-cli/internals/secrethub/command"
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+)
+
+// ServiceGCPLinkCommand create a new link between a SecretHub namespace and a GCP project.
+type ServiceGCPLinkCommand struct {
+	namespace string
+	projectID string
+	io        ui.IO
+	newClient newClientFunc
+}
+
+// NewServiceGCPLinkCommand creates a new ServiceGCPLinkCommand.
+func NewServiceGCPLinkCommand(io ui.IO, newClient newClientFunc) *ServiceGCPLinkCommand {
+	return &ServiceGCPLinkCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+func (cmd *ServiceGCPLinkCommand) Run() error {
+	client, err := cmd.newClient()
+	if err != nil {
+		return err
+	}
+
+	_, err = client.IDPLinks().GCP().Get(cmd.namespace, cmd.projectID)
+	if !api.IsErrNotFound(err) {
+		fmt.Fprintf(cmd.io.Output(), "Namespace %s and GCP project %s are already linked.\n", cmd.namespace, cmd.projectID)
+		return nil
+	}
+
+	return createGCPLink(client, cmd.io, cmd.namespace, cmd.projectID)
+}
+
+func (cmd *ServiceGCPLinkCommand) Register(r command.Registerer) {
+	clause := r.Command("link", "Create a new link between a namespace and a GCP project to allow creating SecretHub service accounts for GCP Service Accounts in the GCP project.")
```

Comment:
Maybe we can explain in a `clause.HelpLong` why a link is required?

Reply:
And shall we add that a GCP project can be linked to multiple namespaces and that a namespace can be linked to multiple GCP projects?

Reply:
Added quite a long `HelpLong`

Reply:
Sweet! Really nice